code/datums/components/engraved.dm

```diff
@@ -0,0 +1,92 @@
+/**
+ * # engraved component!
+ *
+ * component for walls that applies an engraved overlay and lets you examine it to read a story (+ art element yay)
+ * new creations will get a high art value, cross round scrawlings will get a low one.
+ * MUST be a component, though it doesn't look like it. SSPersistence demandeth
+ */
+/datum/component/engraved
+	///the generated story string
+	var/engraved_description
+	///whether this is a new engraving, or a persistence loaded one.
+	var/persistent_save
+	///what random icon state should the engraving have
+	var/icon_state_append
+	///The story value of this piece.
+	var/story_value
+
+/datum/component/engraved/Initialize(engraved_description, persistent_save, story_value)
+	. = ..()
+	if(!isclosedturf(parent))
+		return COMPONENT_INCOMPATIBLE
+	var/turf/closed/engraved_wall = parent
+
+	SSpersistence.wall_engravings += src
+	ADD_TRAIT(engraved_wall, NOT_ENGRAVABLE, TRAIT_GENERIC)
+	src.engraved_description = engraved_description
+	src.persistent_save = persistent_save
+	src.story_value = story_value
+
+
+	var/beauty_value
+	switch(story_value)
+		if(STORY_VALUE_SHIT)
+			beauty_value = rand(-50, 50) //Ugly or mediocre at best
+		if(STORY_VALUE_NONE)
+			beauty_value = rand(0, 100) //No inherent value
+		if(STORY_VALUE_MEH)
+			beauty_value = rand(100, 200) //Its an okay tale
+		if(STORY_VALUE_OKAY)
+			beauty_value = rand(150, 300) //Average story! most things are like this
+		if(STORY_VALUE_AMAZING)
+			beauty_value = rand(300, 600)//Really impactful stories, seeing a lost limb, losing a loved pet.
+		if(STORY_VALUE_LEGENDARY)
+			beauty_value = rand(500, 800) //Almost always a good story! this is for memories you can barely ever get, killing megafauna, doing ultimate feats!
+
+	engraved_wall.AddElement(/datum/element/art, beauty_value / ENGRAVING_BEAUTY_TO_ART_FACTOR)
+	if(persistent_save)
+		engraved_wall.AddElement(/datum/element/beauty, beauty_value)
+	else
+		engraved_wall.AddElement(/datum/element/beauty, beauty_value / ENGRAVING_PERSISTENCE_BEAUTY_LOSS_FACTOR) //Old age does them harm
+	icon_state_append = rand(1, 2)
+	//must be here to allow overlays to be updated
+	RegisterSignal(parent, COMSIG_ATOM_UPDATE_OVERLAYS, .proc/on_update_overlays)
+	engraved_wall.update_appearance()
+
+/datum/component/engraved/Destroy(force, silent)
+	. = ..()
+	var/turf/closed/engraved_wall = parent
+	SSpersistence.wall_engravings -= src
+	REMOVE_TRAIT(engraved_wall, NOT_ENGRAVABLE, TRAIT_GENERIC)
```

Comment:
All this should be in unregister from parent ? And part of Initialize should in Register. Otherwise you're not really supporting proper component transfer.

Reply:
i hope i moved enough